### PR TITLE
Adding ProGuard, dSYM, & Source Maps API docs

### PR DESCRIPTION
**Pull request by rivkahstandig3636**

### CODE REVIEW

**Comment by rivkahstandig3636** [comment]
[ch15802]

**Comment by brianr** [comment]
@rivkahstandig3636 Can you format these to look similar to the other API docs? I was expecting to see something looks like this: https://github.com/rollbar/docs/blob/master/api/deploys.md

File: api/dsym.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+:	your `post_server_item` access token
+
+version
+:	a string indicating the current code version.
+
+bundle_identifier
+:	your bundle identifier
+
+dsym
+:	path to your zipped dSYM file. See [here](https://raw.githubusercontent.com/rollbar/rollbar-ios/master/upload_dsym.py) for more info on this.
```
Comment (brianr) [inline api/dsym.md]: As far as the API is concerned, this needs to be the file itself, not the path to the file. I'm not sure what the best way to word that is - @coryvirok maybe you have a suggestion?

File: api/proguard.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
+To upload the `mapping.txt` file generated by ProGuard, call the `api/1/proguard` endpoint with the following params:
+
+access_token
+:	your `post_server_item` access token
+
+version
+:	a string indicating the current code version. This must match the `android:versionName` in your app's `AndroidManifest.xml`, corresponding to 	the version the `mapping.txt` was generated for.
+
+mapping
+:	the path to `mapping.txt`
```
Comment (brianr) [inline api/proguard.md]: same comment as about dsym

File: api/sourcemaps.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+:	a project-level access token
+
+version
+:	a string indicating the current code version
+
+minified_url
+:	the full URL of the minified file. This should start with `http:` or `https:`, which we'll strip off.
+
+source_map
+:	the URL for the source map file
```
Comment (brianr) [inline api/sourcemaps.md]: same comment as about dsym - this needs to be the actual source map file, not the url to it

File: api/sourcemaps.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
+These calls require a project-level access token, which should be provided in the query string. The prefix for all URLs is `https://api.rollbar.com`.
```
Comment (brianr) [inline api/sourcemaps.md]: In all cases it needs to be a `post_server_item` access token

File: api/sourcemaps.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+To upload your source maps at the beginning of your deploy script (before the new code is in production), call the `api/1/sourcemap` endpoint with the following params:
+
+access_token
+:	a project-level access token
+
+version
+:	a string indicating the current code version
+
+minified_url
+:	the full URL of the minified file. This should start with `http:` or `https:`, which we'll strip off.
```
Comment (brianr) [inline api/sourcemaps.md]: How about:

> The full URL of the minified file, as it appears in the stack trace. This should start with `http:` or `https:`.

File: api/sourcemaps.md [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+source_map
+:	the URL for the source map file
+
+other params 
+:	as needed for your source tree.
```
Comment (brianr) [inline api/sourcemaps.md]: This... doesn't seem very meaningful. Maybe call it out as a separate section, explaining what they keys mean and what the values should be. (key = source file paths, value = the source files themselves).

File: api/sourcemaps.md [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+
+```
+example/
+example/static/js/example.min.js
+example/static/js/example.min.map
+example/static/js/site.js
+example/static/js/util.js
+```
+
+# Download
```
Comment (brianr) [inline api/sourcemaps.md]: I would make this title more specific; as written it reads to me like the call would be to download the source map. This is actually "trigger an automatic download".

File: api/sourcemaps.md [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+Call the `api/1/sourcemap/download` endpoint with the following params:
+
+access_token
+:	a project-level access token
+
+version
+:	a string indicating the current code version
+
+minified_url
+:	the full URL of the minified file. This should start with `http:` or `https:`, which we'll strip off.
```
Comment (brianr) [inline api/sourcemaps.md]: same comment as for minified_url above

**Comment by rivkahstandig3636** [comment]
@brianr Better? :)

File: api/dsym.md [hunk 9]
```diff
@@ -0,0 +1,23 @@
+## Upload your .dSYM files
+
+	POST /api/1/dsym
+
+Requires `post_server_item` scope.
```
Comment (brianr) [inline api/dsym.md]: I think you can drop this since you cover it in the Form Data Parameters section

File: api/dsym.md [hunk 10]
```diff
@@ -0,0 +1,23 @@
+## Upload your .dSYM files
```
Comment (brianr) [inline api/dsym.md]: How about `Upload a .dSYM file`

File: api/dsym.md [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+
```
Comment (brianr) [inline api/dsym.md]: Somewhere in here (either as the description for the call, which would go here, or in the Example section), can we have some text explaining that the way this is usually used is via the upload_dsym script, with a link to the instructions for how to set that up?

I do see the link in the params table, but I think it's important enough to call it out in the main section.

File: api/proguard.md [hunk 12]
```diff
@@ -0,0 +1,22 @@
+## Upload your `mapping.txt` file
```
Comment (brianr) [inline api/proguard.md]: How about `Upload a ProGuard Mapping file`

File: api/proguard.md [hunk 13]
```diff
@@ -0,0 +1,22 @@
...
+
+	POST /api/1/proguard/
```
Comment (brianr) [inline api/proguard.md]: extra `/` at the end

File: api/proguard.md [hunk 14]
```diff
@@ -0,0 +1,22 @@
...
+
+Requires `post_server_item` scope.
+
+### Form Data Parameters
+
+Name | Type | Description
+-----|------|-------------
+`access_token`|`string`|**Required** A `post_server_item`-scope project access token.
+`version`|`string`|**Required** The current code version. This must match the `android:versionName` in your app's `AndroidManifest.xml`, corresponding to the version the `mapping.txt` was generated for.
+`mapping`|`file name`|**Required** Your `mapping.txt` file.
```
Comment (brianr) [inline api/proguard.md]: how about "file upload" for the type.

File: api/dsym.md [hunk 15]
```diff
@@ -0,0 +1,23 @@
...
+Requires `post_server_item` scope.
+
+### Form Data Parameters
+
+Name | Type | Description
+-----|------|-------------
+`access_token`|`string`|**Required** A `post_server_item`-scope project access token.
+`bundle_identifier`|`string`|**Required** The current code version.
+`dsym`|`file name`|**Required** Your zipped dSYM file. See [here](https://raw.githubusercontent.com/rollbar/rollbar-ios/master/upload_dsym.py) for more info.
```
Comment (brianr) [inline api/dsym.md]: how about "file upload" for the type

File: api/sourcemaps.md [hunk 16]
```diff
@@ -0,0 +1,55 @@
...
+
+	POST /api/1/sourcemap/
+
+### Form Data Parameters
+
+Name | Type | Description
+`access_token`|`string`|**Required** A project access token.
+`version`|`string`|**Required** The current code version.
+`minified_url`|`string`|**Required** The full URL of the minified file, as it appears in the stack trace. This should start with `http:` or `https:`.
+`source_map`|`file name`|**Required** Your source map file.
```
Comment (brianr) [inline api/sourcemaps.md]: how about "file upload" for the type (here and below)

File: api/sourcemaps.md [hunk 17]
```diff
@@ -0,0 +1,55 @@
+## Upload
+
+	POST /api/1/sourcemap/
+
+### Form Data Parameters
+
+Name | Type | Description
+`access_token`|`string`|**Required** A project access token.
```
Comment (brianr) [inline api/sourcemaps.md]: a `post_server_item`-scope project access token.

File: api/sourcemaps.md [hunk 18]
```diff
@@ -0,0 +1,55 @@
+## Upload
+
+	POST /api/1/sourcemap/
```
Comment (brianr) [inline api/sourcemaps.md]: extra `/`

File: api/sourcemaps.md [hunk 19]
```diff
@@ -0,0 +1,55 @@
...
+
+```
+example/
+example/static/js/example.min.js
+example/static/js/example.min.map
+example/static/js/site.js
+example/static/js/util.js
+```
+
+# Trigger An Automatic Download
```
Comment (brianr) [inline api/sourcemaps.md]: why is this a `#`-style header when the one above is `##` ?

File: api/sourcemaps.md [hunk 20]
```diff
@@ -0,0 +1,55 @@
...
+
+	POST /api/1/sourcemap/download
+
+### Form Data Parameters
+
+Name | Type | Description
+`access_token`|`string`|**Required** A project access token.
```
Comment (brianr) [inline api/sourcemaps.md]: need to specify the scope

**Comment by rivkahstandig3636** [comment]
@brianr Okay how does this look now?

File: api/dsym.md [hunk 21]
```diff
@@ -0,0 +1,23 @@
...
+`dsym`|`file upload`|**Required** Your zipped dSYM file. See [here](https://raw.githubusercontent.com/rollbar/rollbar-ios/master/upload_dsym.py) for more info.
+
+### Example
+
+```bash
+curl -X POST "https://api.rollbar.com/api/1/dsym"
+	-F access_token=POST_SERVER_ITEM_ACCESS_TOKEN \
+	-F version=0.1.2 \
+	-F bundle_identifier="com.apple.xcode.dsym.org.rollbar.DelightfulApp" \
+	-F dsym=dsym.zip \
```
Comment (brianr) [inline api/dsym.md]: there shouldn't be a `\` at the end of the last line. (that's the line continuation character - used to split commands across lines.)

Comment (rivkahstandig3636) [inline api/dsym.md]: good catch!

**Pull request merged**